src/NuGet.Clients/PackageManagement.VisualStudio/Migrate/ProjectJsonToPackageRefMigrator.cs

```diff
@@ -96,32 +108,28 @@ private static void MigrateRuntimes(PackageSpec packageSpec, Microsoft.Build.Eva
             }
 
             var union = string.Join(";", runtimeIdentifiers.Union(runtimeSupports));
-            NuGetUIThreadHelper.JoinableTaskFactory.Run(async () =>
-            {
-                await NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
-                buildProject.SetProperty("RuntimeIdentifiers", union);
-            });            
+
+            await NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
+            buildProject.SetProperty("RuntimeIdentifiers", union);
         }
 
-        private static void RemoveProjectJsonReference(Microsoft.Build.Evaluation.Project buildProject, string projectJsonFilePath)
+        private static async Task RemoveProjectJsonReferenceAsync(Microsoft.Build.Evaluation.Project buildProject, string projectJsonFilePath)
         {
-            NuGetUIThreadHelper.JoinableTaskFactory.Run(async () =>
-            {
-                await NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
-                var projectJsonItem = buildProject.GetItems("None")
+            await NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
+            var projectJsonItem = buildProject.GetItems("None")
                 .Where(t => string.Equals(t.EvaluatedInclude, Path.GetFileName(projectJsonFilePath), StringComparison.OrdinalIgnoreCase))
                 .FirstOrDefault();
 
-                if (projectJsonItem != null)
-                {
-                    buildProject.RemoveItem(projectJsonItem);
-                }
-            });
+            if (projectJsonItem != null)
+            {
+                buildProject.RemoveItem(projectJsonItem);
+            }            
         }
 
-        private static void CreateBackup(LegacyCSProjPackageReferenceProject project,
+        private static async Task CreateBackupAsync(LegacyCSProjPackageReferenceProject project,
             string projectJsonFilePath)
         {
+            await TaskScheduler.Default;
```

Comment:
Here is the only place you need to hop on a background thread. I'd suggest to wrap all statements here in 
```cs
try{...}
finally
{
  await NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
}
```

instead of switching to UI thread in every other private method.